pkgs/development/python-modules/tiler/default.nix

```diff
@@ -17,8 +20,19 @@ buildPythonPackage rec {
     hash = "sha256-2HWO/iJ9RCWNVmw2slu9F/+Mchk3evB5/F8EfbuMI/Y=";
   };
 
+  patches = [
+    # https://github.com/the-lay/tiler/pull/24
+    (fetchpatch {
+      name = "unpin-setuptools-scm-dependency.patch";
+      url = "https://github.com/the-lay/tiler/commit/7a9f7e32c5f9c263c1ae28bfd19c7539556684cb.patch";
+      hash = "sha256-TMr3LJtiKUxJv2pAzAd8CWs3AtWsF0YS79NzKBN5TKM=";
+    })
+  ];
+
   nativeBuildInputs = [
     setuptools
+    setuptools-scm
+    wheel
```

Comment:
I don't know much about how `build` works, but do we need `wheel` for nativeBuildInputs? Your patch seems to remove it from pyproject.toml.

Reply:
Yes, it looks confusing, and I can explain. We use `build` to build a wheel (instead of an sdist), and `setuptools` always needs `wheel` in order to do this. [PEP 517](https://peps.python.org/pep-0517/#get-requires-for-build-wheel) provides a way for a build backend like setuptools to specify any additional dependencies it needs in order to build a wheel, which setuptools does [here](https://github.com/pypa/setuptools/blob/71b355ba8fcbb9341364699375ee786100b3cf75/setuptools/build_meta.py#L354), so it does not need to be listed in pyproject.toml. (In fact, when it is listed in pyproject.toml, it is always included, even for an sdist build, which is unnecessary in this case.)

A normal PEP 517 build would create an isolated virtual environment and download wheel automatically, but in nixpkgs we have to explicitly specify it.

Reply:
Thanks for the clarification!
I'll take another look at your amazing work in light of this.